lib/galaxy/datatypes/binary.py

```diff
@@ -4302,3 +4309,68 @@ def display_peek(self, dataset: DatasetProtocol) -> str:
             return dataset.peek
         except Exception:
             return "Binary Numpy npz file (%s)" % (nice_size(dataset.get_size()))
+
+
+class FITS(Binary):
+    """
+    FITS (Flexible Image Transport System) file data format, widely used in astronomy
+    Represents sky images (in celestial coordinates) and tables
+    https://fits.gsfc.nasa.gov/fits_primer.html
+    """
+
+    file_ext = "fits"
+
+    MetadataElement(
+        name="HDUs", default=['FITS File HDUs'], desc="Header Data Units", param=metadata.ListParameter, readonly=True,
+        visible=True, no_value=()
+    )
+
+    def __init__(self, **kwd):
+        super().__init__(**kwd)
+        self._magic = b"SIMPLE  ="
+
+    def sniff(self, filename: str) -> bool:
+        """
+        Determines whether the file is a FITS file
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('test.fits')
+        >>> FITS().sniff(fname)
+        True
+        >>> fname = FilePrefix(get_test_fname('interval.interval'))
+        >>> FITS().sniff(fname)
+        False
+        """
+
+        try:
+            # The first 9 bytes of any FITS file are always "SIMPLE  ="
+            with open(filename, "rb") as header:
+                if header.read(9) == self._magic:
+                    return True
+                return False
+        except Exception as e:
+            log.warning("%s, sniff Exception: %s", self, e)
+            return False
+
+    def set_meta(self, dataset: DatasetProtocol, overwrite: bool = True, **kwd) -> None:
+        super().set_meta(dataset, overwrite=overwrite, **kwd)
+        try:
+            with fits.open(dataset.file_name) as hdul:
```

Comment:
if `fits` is not imported it will trigger the exception here, correct? You wanted to have a fallback I assume?

Reply:
Somehow I think @francoismg chooses to implement fallback in `display_peek` if `dataset.peek` is not set. But it's set every time `set_peek` is called. I am not totally sure when is each of these methods called so I do not know what is the behavior this will yield, but I think @francoismg tried it and it worked. So let's see when he works on Monday.

Anyway, I wonder if there should be a test for checking metadata to make sure it's read correctly? Is it possible to make it in both cases, also with fallback?

Reply:
Sure this is possible. Either here as a doc test or even in a tool/integration test should work.

Reply:
If fits is not imported or if there's any other problem while reading the file with astropy it will trigger the exception yes. The "fallback" is implemented by setting a default value to the `MetadataElement` "HDUs"

`MetadataElement(
        name="HDUs", default=['FITS File HDUs'], desc="Header Data Units", param=metadata.ListParameter, readonly=True, visible=True, no_value=()
    )`

So if no specific value for the `MetadataElement` could be set in `set_meta` the default value of "HDUs" will be used in `set_peek` you said to not put too much effort in the fallback so it seemed like the easier way to manage it